### DISCUSSION TITLE
Is the ONLY keyword supported ?

### DISCUSSION THREAD

**Discussion by TeddyLourson** [supabase/supabase#5927] [Questions] [answered]
Hi,
I'm considering handling deletion using partitions (to separate deleted records from non deleted records) as described in this post : 
https://stackoverflow.com/questions/506432/cascading-soft-delete

But once implemented, if I want to be able to select records (only non-deleted ones) I have to SELECT like this with Postgresql :
```
SELECT name, altitude
    FROM ONLY cities
    WHERE altitude > 500;
```
ref. : https://www.postgresql.org/docs/9.1/ddl-inherit.html#:~:text=Here%20the%20ONLY%20keyword%20indicates,DELETE%20%E2%80%94%20support%20the%20ONLY%20keyword.

Is such a call possible using supabase with react ?
How would that translate ?

Thanks in advance !

**Comment by burggraf** [comment, marked as answer]
I assume you're talking about using the Supabase JavaScript API to make this call.  The API uses [Postgrest](https://postgrest.org/en/stable/), and while I could not find any references there to indicate that it's supported directly, you could put your SQL code with the `ONLY` keyword into a PostgreSQL function and call it with `.rpc()`: see [Postgres functions: rpc()](https://supabase.com/docs/reference/javascript/rpc).

**Reply by steve-chavez** [reply]
@Teio07 I suggest you file an issue on https://github.com/PostgREST/postgrest/issues, seems like something that can be added.

**Reply by TeddyLourson** [reply]
I'll do that. In the meanwhile, I'll use RPC and once it's available from the JS API, I'll switch to it.
Here's the issue : https://github.com/supabase/supabase/issues/5932
